File: 0001-two-sum/0001-two-sum.cpp
class Solution {
private:
    int findIndex(vector<int>& nums, int num, int c) {
        for (int i = 0; i < nums.size(); i++) {
            if (num == nums[i] && i != c) {
                return i;
            }
        }
        return 0;
    }

public:
    vector<int> twoSum(vector<int>& nums, int target) {
        int left = 0, right = nums.size() - 1;
        vector<int> temp = nums;
        sort(nums.begin(), nums.end());
        while (left < right) {
            if (nums[left] + nums[right] > target) {
                right--;
            } else if (nums[left] + nums[right] < target) {
                left++;
            } else {
                break;
            }
        }
        return {findIndex(temp, nums[left], -1),
                findIndex(temp, nums[right], findIndex(temp, nums[left], -1))};
    }
};

File: 0001-two-sum/0001-two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function (nums, target) {
    let left = 0, right = nums.length - 1, temp = 0, res = [];
    const sorted = [...nums].sort(function (a, b) { return a - b });
    function findRight(indexLeft, sortedRight) {
        for (let i = 0; i < nums.length; i++) {
            if (sortedRight === nums[i] && indexLeft !== i) {
                return i;
            }
        }
    }
    while (left <= right) {
        temp = sorted[left] + sorted[right];
        if (target === temp) {
            const indexLeft = nums.indexOf(sorted[left]);
            res.push(indexLeft);
            res.push(findRight(indexLeft, sorted[right]));
            return res;
        }

        if (target < temp) {
            right--;
        }

        if (target > temp) {
            left++;
        }
    }
};